plugins/catalog-backend-module-azure-blobstorage/src/provider/AzureBlobStorageEntityProvider.test.ts

```diff
@@ -0,0 +1,314 @@
+  private tasks: TaskInvocationDefinition[] = [];
+
+  getTasks() {
+    return this.tasks;
+  }
+
+  run(task: TaskInvocationDefinition): Promise<void> {
+    this.tasks.push(task);
+    return Promise.resolve(undefined);
+  }
+}
+
+const keys = ['key1.yaml', 'key2.yaml', 'key3.yaml', 'key 4.yaml'];
+
+jest.mock('@azure/storage-blob', () => {
+  class ContainerClient {
+    constructor(private readonly containerName: string) {}
+
+    listBlobsFlat(options: ContainerListBlobsOptions) {
+      if (this.containerName === 'container' && options.prefix === 'sub/dir/') {
+        return [
+          {
+            name: `${options.prefix}${keys[0]}`,
+          },
+          {
+            name: `${options.prefix}${keys[1]}`,
+          },
+          {
+            name: `${options.prefix}${keys[2]}`,
+          },
+          {
+            name: `${options.prefix}${keys[3]}`,
+          },
+        ];
+      }
+      return [{}];
+    }
+  }
+
+  class BlobServiceClient {
+    constructor(public readonly url: string) {}
+
+    getContainerClient(containerName: string) {
+      return new ContainerClient(containerName);
+    }
+  }
+
+  return {
+    __esModule: true,
+    BlobServiceClient,
+  };
+});
+
+const logger = getVoidLogger();
+jest.spyOn(logger, 'error').mockReturnValue(logger);
+
+describe('azureBlobStorageEntityProvider', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const expectMutation = async (
+    providerId: string,
+    providerConfig: object,
+    expectedBaseUrl: string,
+    names: Record<string, string>,
+    integrationConfig?: object,
+    scheduleInConfig?: boolean,
+  ) => {
+    const config = new ConfigReader({
+      integrations: {
+        azureBlobStorage: integrationConfig ? [integrationConfig] : [],
+      },
+      catalog: {
+        providers: {
+          azureBlobStorage: {
+            [providerId]: providerConfig,
+          },
+        },
+      },
+    });
+
+    const schedulingConfig: Record<string, any> = {};
+
+    const schedule = new PersistingTaskRunner();
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+
+    if (scheduleInConfig) {
+      schedulingConfig.scheduler = {
+        createScheduledTaskRunner: (_: any) => schedule,
+      } as unknown as PluginTaskScheduler;
+    } else {
+      schedulingConfig.schedule = schedule;
+    }
+
+    const provider = AzureBlobStorageEntityProvider.fromConfig(config, {
+      ...schedulingConfig,
+      logger,
+    })[0];
+    expect(provider.getProviderName()).toEqual(
+      `azureBlobStorage-provider:${providerId}`,
+    );
+
+    await provider.connect(entityProviderConnection);
+
+    const taskDef = schedule.getTasks()[0];
+    expect(taskDef.id).toEqual(
+      `azureBlobStorage-provider:${providerId}:refresh`,
+    );
+    await (taskDef.fn as () => Promise<void>)();
+
+    const expectedEntities = keys.map(key => {
+      // @ts-ignore
```

Comment:
would be really nice to try to avoid ts-ignore entirely everywhere - it's very heavyhanded and disables way more than one normally wants